system/src/system_listening_mode.cpp

```diff
@@ -51,11 +48,35 @@ const auto SETUP_UPDATE_INTERVAL = 1000;
 } // unnamed
 
 ListeningModeHandler::ListeningModeHandler()
-        : active_(false) {
+        : active_(false)
+#if HAL_PLATFORM_BLE
```

Comment:
Let's not pollute this class like this with BLE-specific stuff and decouple all of this into a separate method/class e.g. `BleListneningModeHandler`.